tools/pd-ctl/pdctl/command/component_command.go

```diff
@@ -112,15 +112,29 @@ func deleteComponentConfigCommandFunc(cmd *cobra.Command, args []string) {
 }
 
 func getComponentIDCommandFunc(cmd *cobra.Command, args []string) {
-	if len(args) != 1 {
+	argsLen := len(args)
+	// args too long
+	if argsLen > 1 {
 		cmd.Usage()
 		return
 	}
+	var (
+		hasIdsVal = false
+		ids       = "all"
```

Comment:
How about printing all ids when there is no argument?

Reply:
In `server/api/router.go`
```suggestion
				if varName == "all" {
			rd.JSON(w, http.StatusOK, svr.GetConfigManager().GetAllComponentIDs())
			} else {
				rd.JSON(w, http.StatusOK, svr.GetConfigManager().GetComponentIDs(varName))
			}
```

Reply:
What I mean is that we may not need to use "all" here. If `vars["component"]` is empty, we can directly call `GetAllComponentIDs`.

Reply:
Thank you for reminding me. I'll change it later.

Reply:
If don't set a flag, other route captures will occur, such as:
```suggestion
	componentRouter.Handle("/{component_id}", gwmux).Methods("GET")
```
Do you have any good ideas?

Reply:
`ids` subcommand will be captured by `/component/ids/{component}` rather than `/{component_id}`

Reply:
However, after local debugging, I found that when my`RequestURI` = `/pd/api/v1/component/ids `, it is actually handled by handleComponentGet function.

Reply:
Oh, sorry. Just keep it.